pyannote/audio/core/task.py

```diff
@@ -232,14 +237,27 @@ def setup(self, stage=None):
         If `specifications` attribute has not been set in `__init__`,
         `setup` is your last chance to set it.
         """
-        pass
 
     def setup_loss_func(self):
         pass
 
     def setup_validation_metric(self):
         pass
 
+    def _set_augmentation_sample_rate(self, augmentation: BaseWaveformTransform):
```

Comment:
This feels like a hack -- I'd rather not have this kind of logic within `Task` class.
I'd rather not touch `augmentation` internals from within `Task`.

Furthermore, what if we want to support another augmentation library later? 
This will be a nighmare to maintain...

Hence my original proposal to let the augmentation registry handle this.
We'd only have to write another `@singledispatch` entry in `registry`.